src/main/java/com/exasol/mavenpluginintegrationtesting/MavenIntegrationTestEnvironment.java

```diff
@@ -0,0 +1,178 @@
+package com.exasol.mavenpluginintegrationtesting;
+
+import java.io.*;
+import java.nio.file.*;
+import java.util.logging.Logger;
+
+import org.apache.maven.it.VerificationException;
+import org.apache.maven.it.Verifier;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.apache.maven.shared.utils.io.FileUtils;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+
+import com.exasol.errorreporting.ExaError;
+
+/**
+ * This class sets up an isolated local maven repository for testing maven plugins without installing them.
+ */
+public class MavenIntegrationTestEnvironment {
+    public static final String COVERAGE_KEY = "test.coverage";
+    public static final String DEBUG_KEY = "test.debug";
+    private static final Logger LOGGER = Logger.getLogger(MavenIntegrationTestEnvironment.class.getName());
+    static Path mavenRepo;
+    private static int jacocoAgentCounter = 0;
+
+    /**
+     * Create a new instance of {@link MavenIntegrationTestEnvironment}.
+     */
+    public MavenIntegrationTestEnvironment() {
+        createTemporaryLocalRepositoryIfNotExist();
+        printDebuggerWarning();
+    }
+
+    private static void createTemporaryLocalRepositoryIfNotExist() {
+        mavenRepo = Path.of(System.getProperty("java.io.tmpdir"))
+                .resolve("project-keeper-integration-test-maven-repository");
+        if (!mavenRepo.toFile().exists()) {
+            mavenRepo.toFile().mkdir();
+        }
+    }
+
+    private static synchronized int getNextJacocoAgentNumber() {
+        final int thisNumber = jacocoAgentCounter;
+        jacocoAgentCounter++;
+        return thisNumber;
```

Comment:
`return jacocoAgentCounter++;` would also work I think

Reply:
yea but @redcatbear doesen't liked it in a review